channels/telegram/src/main/kotlin/com/justai/jaicf/channel/telegram/TelegramReactions.kt

```diff
@@ -18,11 +19,34 @@ class TelegramReactions(
 ) : Reactions() {
 
     val chatId = request.chatId
+    private val messages = mutableListOf<Message>()
+
+    private fun addResponse(pair: Pair<retrofit2.Response<Response<Message>?>?, Exception?>) {
+        pair.first?.body()?.result?.let { messages.add(it) }
+    }
 
     override fun say(text: String): SayReaction {
         return sendMessage(text)
     }
 
+    override fun buttons(vararg buttons: String): ButtonsReaction {
+        messages.lastOrNull()?.let { message ->
+            val keyboard = message.replyMarkup?.inlineKeyboard?.toMutableList() ?: mutableListOf()
+            keyboard.addAll(buttons.map { listOf(InlineKeyboardButton(it, callbackData = it)) })
+
+            api.editMessageReplyMarkup(
+                    chatId,
+                    message.messageId,
+                    replyMarkup = InlineKeyboardMarkup(keyboard)
+            ).also {
+                messages.remove(message)
```

Comment:
Maybe change it to `messages.removeIf { it.messageId == message.messageId }`?
`Message` is a huge data class with a lot fields, nested data classes, etc.
I'd not rely on Message's `equals`